Document k6/experimental/redis module (#768)
grafana/k6-docs: src/data/markdown/docs/02 javascript api/13 k6-redis.md

---
title: "k6/experimental/redis"
excerpt: "k6 Redis experimental API"
---

<Blockquote mod='info'>

#### This module is experimental, use at your own risk

**This module is experimental**. It is not guaranteed to be stable, and may change without notice. **Use at your own risk**.
While will we intend to keep this module as simple and stable as possible, we may need to add more features or introduce breaking changes in the future, and won't refrain from doing so until we release this module as a "stable" one. Please kindly provide us your user feedback, and open an issue or pull request if you have any suggestions.
</Blockquote>

> The [Redis](https://redis.io/) module for [k6](https://github.com/grafana/k6) provides a client library allowing users to interact with Redis, directly from their k6 scripts. It was built with the objective to enable users to either load test their Redis infrastructure, or use Redis as a data store backing their load tests' script logic.

Reviewer: ```suggestion
The [Redis](https://redis.io/) module for [k6](https://github.com/grafana/k6) provides a client library to interact with Redis, directly from their k6 scripts.
Its objective is to let users to either load test their Redis infrastructure, or use Redis as a data store for their test-script logic.
```

Reviewer: I don't have a strong reason to remove "load". I did it for a few reasons:
- k6 could be for general performance testing, too
- it shortens the sentence
- I think style guidelines would say it needed to be "load-test-script logic", and while this would be very pleasing to me, I think it is too fussy for our docs.

Author: The first sentence in your suggestion should drop the "their" from "directly from their k6 scripts". As the users it referred to is now gone 😄

> The module's API intends to be quite thorough and extensive, but is still limited to a set of functionalities the k6 team judged useful in the context of k6. The intent of this module isn't to provide a full-fledged Redis client, integrating the whole Redis API, but rather to empower our users to use Redis in use-cases we believe are relevant to k6. 

Reviewer: ```suggestion
Though the API intends to be quite thorough and extensive, it is still limited to a set of functionalities that the k6 team judged useful in the context of k6.
Instead of a full-fledged Redis client, integrating the whole Redis API, the modules aims to empower users to use Redis for use cases most appropriate to k6. 
```

This module is **not stable**, and remains open to changes. Thus, if any Redis functionalities were to be missing for your use-case, feel free to open an issue or pull request.

### Notes on usage

> The `Client` exposes a [promise](https://javascript.info/promise-basics)-based API. Unlike most other current k6 modules and extensions, who operate in a synchronous manner,

Reviewer: ```suggestion
The `Client` exposes a [promise](https://javascript.info/promise-basics)-based API.
Unlike most other current k6 modules and extensions,
which operate in a synchronous manner,
```

> the Redis `Client` operates in an asynchronous manner. In practice, this means that using the Redis `Client`'s methods won't block the execution of the test, and that the test will continue to run even if the Redis `Client` is not ready to respond to the request.

Reviewer: ```suggestion
the Redis `Client` operates in an asynchronous manner.
In practice, this means that using the Redis `Client`'s methods won't block test execution,
and that the test will continue to run even if the Redis `Client` isn't ready to respond to the request.
```

> One potential caveat introduced by this new execution model, is that to depend on operations against Redis, all following operations should be made in the context of a [promise chain](https://javascript.info/promise-chaining). As demonstrated in the examples below, whenever there is a dependency on a Redis operation result or return value, the operation, should be wrapped in a promise itself. That way, a user can perform asynchronous interactions with Redis in a seemingly synchronous manner.

Reviewer: ```suggestion
This new execution model does introduce a potential caveat: to depend on operations against Redis, all following operations should be made in the context of a [promise chain](https://javascript.info/promise-chaining).
As the examples below demonstrate, whenever there is a dependency on a Redis operation result or return value, the operation should be wrapped in a promise itself.
This way, a user can perform asynchronous interactions with Redis in a seemingly synchronous manner.
```

## API

| Class                                       | Description                                                                                    |
| :------------------------------------------ | :--------------------------------------------------------------------------------------------- |

> | [Client](/javascript-api/k6-redis/client)   | Client exposing allowing interactions with Redis.                                              |

Reviewer: > Client exposing allowing interactions with Redis.

Not sure here. Maybe:

Client that exposes allowed interactions with Redis.

Author: You're right, direct form is probably better here indeed!

---
title: "k6/experimental/redis"
excerpt: "k6 Redis experimental API"
---

<Blockquote mod="attention" title="The redis module is experimental, use at your own risk">

While we intend to keep this module as simple and stable as possible,
we may need to add features or introduce breaking changes.
This could happen at any time until we release this module as stable.

**Use at your own risk!**

Feel free to provide user feedback, and open an issue or pull request if you have any suggestions.

</Blockquote>

> The [Redis](https://redis.io/) module for [k6](https://github.com/grafana/k6) provides a client library to interact with Redis, directly from their k6 scripts.
> Its objective is to let users to either load test their Redis infrastructure, or use Redis as a data store for test-script logic.

Reviewer: ```suggestion
The [Redis](https://redis.io/) module provides a client library that makes it possible to interact with Redis directly from a k6 script.
With this module, you can load test Redis or use Redis as a data store for test-script logic.
```

Though the API intends to be thorough and extensive, it does not expose the whole Redis API.
Instead, the intent is to expose Redis for use cases most appropriate to k6. 

> This module is **not stable**, and remains open to changes. Thus, if any Redis functionalities were to be missing for your use-case, feel free to open an issue or pull request.

Reviewer: ```suggestion
This module is **not stable**, and remains open to changes. So, if you have a use case for some Redis functionality that's missing, feel free to open an issue or pull request.
```
